R/jaspBivariate.R

```diff
@@ -0,0 +1,256 @@
+#' @param groupName Character; label of the grouping variable displayed as a legend title. If left empty, the name of the \code{group} object is displayed.
+#' @param type Character; How should the distribution of the data be displayed:
+#' \describe{
+#'    \item{"point"}{Using [geom_point].}
+#'    \item{"hex"}{Using [ggplot2::geom_hex].}
+#'    \item{"bin"}{Using [ggplot2::geom_bin2d].}
+#'    \item{"contour"}{Using [ggplot2::geom_density2d].}
+#'    \item{"density"}{Using [ggplot2::geom_density2d_filled].}
+#' }
+#' @param args A list of additional arguments passed to the geom function determined by \code{type} argument.
+#' @param smooth Character; passed as \code{method} argument to [ggplot2::geom_smooth],
+#' unless \code{smooth == "none"}, in which case the layer is not plotted.
+#' @param smoothCi Logical; Should confidence interval around the smooth line be plotted?
+#' Passed as \code{se} argument to [ggplot2::geom_smooth].
+#' @param smoothCiLevel Numeric; Confidence level of the confidence interval around the smooth line.
+#' Passed as \code{level} argument to [ggplot2::geom_smooth].
+#' @param smoothArgs A list of additional arguments passed to [ggplot2::geom_smooth].
+#' @param predict Character; Method for drawing the prediction interval:
+#' \describe{
+#'   \item{"none"}{Prediction interval is not displayed.}
+#'   \item{"lm"}{Prediction interval is plotted, the confidence bands are calculated using [stats::predict.lm].}
+#'   \item{"ellipse"}{Prediction ellipse is plotted using [ggplot2::stat_ellipse].}
+#' }
+#' @param predictLevel Numeric; Confidence level of the prediction interval.
+#' @param predictArgs A list of additional arguments passed to the function that draws the prediction interval.
+#' @param xBreaks Optional numeric vector that specifies the breaks along the x-axis.
+#' @param yBreaks Optional numeric vector that specifies the breaks along the y-axis.
+#' @param legendPosition Character; passed as \code{legend.position} to [themeJaspRaw].
+#' @export
+jaspBivariate <- function(
+    x, y, group = NULL, xName, yName, groupName,
+    type               = c("point", "hex", "bin", "contour", "density", "none"),
+    args               = list(color = "black"),
+    smooth             = c("none", "lm", "glm", "gam", "loess"),
+    smoothCi           = FALSE,
+    smoothCiLevel      = 0.95,
+    smoothArgs         = list(),
+    predict            = c("none", "lm", "ellipse"),
+    predictLevel       = 0.95,
+    predictArgs        = list(),
+    xBreaks            = NULL,
+    yBreaks            = NULL,
+    legendPosition     = "none"
+) {
+
+  type    <- match.arg(type)
+  smooth  <- match.arg(smooth)
+  predict <- match.arg(predict)
+
+  if (is.null(group)) {
+    df  <- data.frame(x = x, y = y)
+    aes <- ggplot2::aes(x = x, y = y)
+  } else {
+    if(type != "point" && type != "none")
+      stop("grouping variable is allowed only for type = 'point' or 'none'.")
```

Comment:
```suggestion
      stop2("grouping variable is allowed only for type = 'point' or 'none'.")
```
basically `stop` with `domain = NA` as default.